Extensions/Physics2Behavior/Box2D_v2.3.1_min.wasm.js

```diff
@@ -7,20 +7,22 @@ function(Box2D) {
   Box2D = Box2D || {};
 
 
-var a;a||(a=typeof Box2D !== 'undefined' ? Box2D : {});var aa;a.ready=new Promise(function(b){aa=b});var ba={},ca;for(ca in a)a.hasOwnProperty(ca)&&(ba[ca]=a[ca]);var da=!1,ea=!1,fa=!1,ha=!1;da="object"===typeof window;ea="function"===typeof importScripts;fa="object"===typeof process&&"object"===typeof process.versions&&"string"===typeof process.versions.node;ha=!da&&!fa&&!ea;var ia="",ja,ka,la,ma;
+var a;a||(a=typeof Box2D !== 'undefined' ? Box2D : {
+  locateFile: function(path, prefix) {
+    return location.protocol === 'file' ?
+        // This is needed to run in preview.
+        prefix + "Extensions/Physics2Behavior/" + path :
+        prefix + path;
+  },
```

Comment:
Do you think it's okay to put it here?

(I only tested a preview and a Liluo build)

Reply:
Careful the protocol contains the `:` too.

Reply:
As for the question, it's a bit dirty and it would be better to have a global initializeBox2D that we can call like `initializeBox2D({ locateFile: ... }).then(Box2D => /*use it/store it somewhere*/)`. But it's ok if it's simpler.

Reply:
What do you mean by "global"?
I think each extension should have their own library. For instance, there is another version of Box2D in the deprecated Physics extension.

I guess, we could move the implementation of `locateFile` at the end of the lib file where the promise is registered, but the inside of the library still needs to be modified a bit.

Reply:
I meant the bit at the end that register the library :) We currently have this:
```
gdjs.registerAsynchronouslyLoadingLibraryPromise(Box2D(window.Box2D).then(box2d => {
  window.Box2D = box2d;
}));
```

I *think* this could be changed to:
```
gdjs.registerAsynchronouslyLoadingLibraryPromise(initializeBox2D({ locateFile: ... }).then(box2d => {
  window.Box2D = box2d;
}));
```

If you basically rename the IIFE that you put around the library:
<img width="1241" alt="image" src="https://user-images.githubusercontent.com/1280130/184087100-c840af00-7e18-498c-9c86-38d40eb81b3d.png">
into `initializeBox2D`:
<img width="1245" alt="image" src="https://user-images.githubusercontent.com/1280130/184087161-7b58ed97-46ac-4821-b7fe-efe5f443f1c0.png">

This should work because what you pass to this IIFE (that I renamed `initializeBox2D` is an object that is used to pass options to the library, including locateFile):
<img width="627" alt="image" src="https://user-images.githubusercontent.com/1280130/184087355-c8d451db-7b3f-4a75-8795-3b6345c41264.png">


So in short, the process should be:
- Take the original Box2D .wasm.js file
- Put it all in a IIFE like done already, but name this IIFE `initializeBox2D` (that's unfortunately a global variable).
- Add the promise registration at the end: 
  ```
  gdjs.registerAsynchronouslyLoadingLibraryPromise(initializeBox2D({locateFile: ... }).then(box2d => {
    window.Box2D = box2d;
  }));
  ```
- That's it! 
- If you are worried about `initializeBox2D` being left in the global namespace, you can consider wrapping the whole thing into another IIFE (`(function() { ... })()`) to avoid leaking it.